index.js

```diff
@@ -48,13 +49,73 @@ module.exports = {
           ), self.findOrCreateUser(spec));
           spec.name = dummy.name;
         }
+        spec.label = spec.label || spec.name;
         spec.options.callbackURL = self.getCallbackUrl(spec, true);
         self.strategies[spec.name] = new Strategy(spec.options, self.findOrCreateUser(spec));
         self.apos.login.passport.use(self.strategies[spec.name]);
         self.addLoginRoute(spec);
         self.addCallbackRoute(spec);
         self.addFailureRoute(spec);
       });
+
+      self.on('apostrophe-login:after', 'redirectToNewLocale', async req => {
```

Comment:
This event has been discussed in A3. In A2 it has a last chance opportunity to set `req.redirect` which is leveraged here.